src/main/java/net/minecraftforge/common/crafting/NBTIngredient.java

```diff
@@ -78,17 +93,18 @@ public JsonElement toJson()
 
         @Override
         public NBTIngredient parse(FriendlyByteBuf buffer) {
-            return new NBTIngredient(buffer.readItem());
+            return new NBTIngredient(buffer.readItem(), buffer.readBoolean());
         }
 
         @Override
         public NBTIngredient parse(JsonObject json) {
-            return new NBTIngredient(CraftingHelper.getItemStack(json, true));
+            return new NBTIngredient(CraftingHelper.getItemStack(json, true), GsonHelper.getAsBoolean(json, "exact_match", false));
         }
 
         @Override
         public void write(FriendlyByteBuf buffer, NBTIngredient ingredient) {
             buffer.writeItem(ingredient.stack);
+            buffer.writeBoolean(ingredient.needsExactMatch());
```

Comment:
This is also a network break. So that's a no go.

Reply:
How do you recommend that I address this? There has to be a boolean, unless you recommend me making a duplicate `forge:nbt_predicate`, or putting the boolean inside the Stack's NBT, which sounds like a terrible solution